descope/auth/enchantedlink.go

```diff
@@ -0,0 +1,100 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/descope/go-sdk/descope/errors"
+)
+
+type enchantedLink struct {
+	authenticationsBase
+}
+
+func (auth *enchantedLink) SignIn(identifier, URI string, r *http.Request, loginOptions *LoginOptions) (*EnchantedLinkResponse, error) {
+	var pswd string
+	var err error
+	if identifier == "" {
+		return nil, errors.NewInvalidArgumentError("identifier")
+	}
+	if loginOptions.IsJWTRequired() {
+		pswd, err = getValidRefreshToken(r)
+		if err != nil {
+			return nil, errors.InvalidStepupJwtError
+		}
+	}
+	httpResponse, err := auth.client.DoPostRequest(composeEnchantedLinkSignInURL(), newMagicLinkAuthenticationRequestBody(identifier, URI, true, loginOptions), nil, pswd)
```

Comment:
Shouldn't it be newEnchantedLinkAuthenticationRequestBody?

Reply:
Left as same body, since those are the same params